File: distros/J/JavonetPerlSdk/lib/Javonet/Binaries/Nodejs/core/handler/InvokeStaticMethodHandler.js
const AbstractHandler = require("./AbstractHandler");

class InvokeStaticMethodHandler extends AbstractHandler {
  requiredParametersCount = 2;

  constructor() {
    super();
  }

  process(command) {
    try {
      if (command.payload.length < this.requiredParametersCount) {
        throw new Error("Invoke Static Method parameters mismatch");
      }
      const { payload } = command;
      let type = payload[0];
      let methodName = payload[1];
      let args = payload.slice(2);
      let method = type[methodName];
      if (typeof method === "undefined") {
        let methods = Object.getOwnPropertyNames(type).filter(function (
          property
        ) {
          return typeof type[property] === "function";
        });
        let message = `Method ${methodName} not found in class. Available methods:\n`;
        methods.forEach((methodIter) => {
          message += `${methodIter}\n`;
        });
        throw new Error(message);
      } else {
        return Reflect.apply(method, undefined, args);
      }
    } catch (error) {
      throw this.process_stack_trace(error, this.constructor.name);
    }
  }
}

module.exports = new InvokeStaticMethodHandler();

File: distros/J/JavonetPerlSdk/lib/Javonet/Binaries/Ruby/Linux/X64/core/handler/invoke_static_method_handler.rb
require_relative 'abstract_command_handler'

class InvokeStaticMethodHandler < AbstractCommandHandler
  def initialize
    @required_parameters_count = 2
  end

  def process(command)
    invoke_static_method(command)
  end

  def invoke_static_method(command)
    begin
      if command.payload.length < @required_parameters_count
        raise ArgumentError.new "Static method parameters mismatch"
      end
  
      method_name = command.payload[1]
      begin
        if command.payload.length > @required_parameters_count
          args = command.payload[2..]
          result = command.payload[0].send(method_name, *args)
        else
          result = command.payload[0].send(method_name)
        end
        return result
      rescue NoMethodError
        methods = command.payload[0].methods
        message = "Method #{method_name} not found in class #{command.payload[0].name}. Available methods:\n"
        methods.each { |method_iter| message += "#{method_iter}\n" }
        raise Exception, message
      end
    rescue Exception => e
      return e
    end
  end
end
